pkg/operation/botanist/dns.go

```diff
@@ -156,6 +176,36 @@ func (b *Botanist) DefaultExternalDNSEntry(seedClient client.Client) component.D
 	))
 }
 
+// DefaultExternalDNSOwner returns the external DNSOwner if external DNS is
+// enabled and if not, DeployWaiter which removes the external DNSOwner
+func (b *Botanist) DefaultExternalDNSOwner(seedClient client.Client) component.DeployWaiter {
+	if b.NeedsExternalDNS() {
+		return dns.NewDNSOwner(
+			&dns.OwnerValues{
+				Name:    DNSExternalName,
+				Active:  true,
+				OwnerID: *b.Shoot.Info.Status.ClusterIdentity,
```

Comment:
This creates a nil pointer panic in the gardenlet. 
Both the care controller and the shoot reconciliation create a new operation.
If the care controller is "quicker"(  e.g Seed is not yet ready etc. - see operation is created [here](https://github.com/gardener/gardener/blob/c0c5a694ae28d90f06c080008f5464d161fb30d3/pkg/gardenlet/controller/shoot/shoot_care_control.go#L175)), compared to the shoot reconciliation (creates the cluster-identity [here](https://github.com/gardener/gardener/blob/f55437b175a3868f220c26514948be6248a425f3/pkg/gardenlet/controller/shoot/shoot_control_reconcile.go#L107-L106)) , the cluster identity in the Shoot status is nil.

Logs
```
E0724 11:20:24.895323   29917 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 568 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x26aa720, 0x3ad6090)
	/Users/d060239/go/src/github.com/gardener/gardener/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/d060239/go/src/github.com/gardener/gardener/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x26aa720, 0x3ad6090)
	/usr/local/Cellar/go/1.14/libexec/src/runtime/panic.go:967 +0x15d
github.com/gardener/gardener/pkg/operation/botanist.(*Botanist).DefaultExternalDNSOwner(0xc001846060, 0x2c30260, 0xc0002645a0, 0x2c1a980, 0xc001e0c720)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/operation/botanist/dns.go:187 +0xf0
github.com/gardener/gardener/pkg/operation/botanist.New(0xc000e8d680, 0x0, 0x0, 0x290c758)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/operation/botanist/botanist.go:80 +0x3c0
github.com/gardener/gardener/pkg/gardenlet/controller/shoot.(*defaultCareControl).Care(0xc0006ad020, 0xc0003a6380, 0xc00126a100, 0x14, 0xc00017dfc0, 0x1)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/gardenlet/controller/shoot/shoot_care_control.go:175 +0x7b7
github.com/gardener/gardener/pkg/gardenlet/controller/shoot.(*Controller).reconcileShootCareKey(0xc00082a6e0, 0xc00126a100, 0x14, 0x1, 0xc00126d610)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/gardenlet/controller/shoot/shoot_care_control.go:80 +0x1c4
github.com/gardener/gardener/pkg/controllerutils.DeprecatedCreateWorker.func1(0xc00126a0e0, 0xa, 0xc00126a0eb, 0x9, 0x0, 0x0, 0xc00126d5e0, 0x0)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/controllerutils/worker.go:46 +0x20a
sigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile(0xc0006a8090, 0xc00126a0e0, 0xa, 0xc00126a0eb, 0x9, 0x9, 0x0, 0x0, 0x0)
	/Users/d060239/go/src/github.com/gardener/gardener/vendor/sigs.k8s.io/controller-runtime/pkg/reconcile/reconcile.go:93 +0x4e
github.com/gardener/gardener/pkg/controllerutils.worker.func1.1(0x2c35840, 0xc000cda300, 0x2bcb5a0, 0xc0006a8090, 0x290145e, 0xa, 0x0)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/controllerutils/worker.go:99 +0x124
github.com/gardener/gardener/pkg/controllerutils.worker.func1()
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/controllerutils/worker.go:116 +0x7e
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000a401c0)
	/Users/d060239/go/src/github.com/gardener/gardener/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5f
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a401c0, 0x3b9aca00, 0x0, 0x1, 0xc0007d4240)
	/Users/d060239/go/src/github.com/gardener/gardener/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/Users/d060239/go/src/github.com/gardener/gardener/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
github.com/gardener/gardener/pkg/controllerutils.CreateWorker.func1(0x2c35840, 0xc000cda300, 0x290145e, 0xa, 0x2bcb5a0, 0xc0006a8090, 0x2c1a600, 0xc00099c600, 0xc000051860, 0xc00068eda0)
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/controllerutils/worker.go:57 +0xb4
created by github.com/gardener/gardener/pkg/controllerutils.CreateWorker
	/Users/d060239/go/src/github.com/gardener/gardener/pkg/controllerutils/worker.go:56 +0xec
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x246e210]
```

Reply:
I was able to reproduce this issue as well. I think it's reproduced very reliably if you create a new shoot instead of migrating an existing one. We brainstormed it a bit with @plkokanov and concluded that in order to fix it, the DNS owner should be inititalized later on, after creating the API server load balancer, similarly to the DNS entry. I will provide a fix shortly.